### ISSUE TITLE
Possible problems about the training dataset

### ISSUE THREAD

**Issue by FlyingDutchman26** [MARIO-Math-Reasoning/Super_MARIO#15]
![image](https://github.com/MARIO-Math-Reasoning/Super_MARIO/assets/110508813/930a88b1-0459-4848-bdf6-86b97fa61c86)
![image](https://github.com/MARIO-Math-Reasoning/Super_MARIO/assets/110508813/5f79ea9c-0984-41a0-9d46-8df6509c290d)
hello! Thanks for your work.
I have some problems with your training data in huggingface.
It seems that some negative examples are not exactly negative? They just have incorrect output format. But their 'Q' is -1.0
Above are some examples I saw in your huggingface dataset.

Moreover, I wonder what  'P' means in your dataset, if 'Q' means the reward value? Could you please explain for me.

Thanks a lot! Best wishes.

**Comment by lovecambi** [comment]
Thanks for your question. 

> It seems that some negative examples are not exactly negative? 

You're correct. When calculating the reward in the MCTS, to speedup this process, we use rule-based method to extract the predicted final answers and check the equivalence with ground truth. This is not 100% correct. If you want to achieve better accuracy, one option is to leverage LLM to extract and recheck the incorrect final answers by rule in the MCTS process. But it actually requires to manipulate two LLMs, one for solution generation, another for answer re-checking. A simple and efficient way is to filter out these false negative examples in the post-processing.

The rule is rigorous such that the only positive example can contribute to the SFT loss. It won't affect the policy training. These data may affect the training of value model, but the influence should be limited.

> Moreover, I wonder what 'P' means in your dataset, if 'Q' means the reward value? Could you please explain for me.

`Q` of intermediate node literally represents the `Q`-value in MCTS, and `Q` of terminal node equals to the calculated reward. So `Q` is the label of value loss.

`P` is the [prior probability](https://github.com/MARIO-Math-Reasoning/Super_MARIO/blob/main/mcts_math/agents/mcts.py#L124).

**Comment by FlyingDutchman26** [comment]
Thanks for your explanation!
